src/wazuh.py

```diff
@@ -91,7 +91,10 @@ def configure_git(
         custom_config_repository: the git repository to add to known hosts.
         custom_config_ssh_key: the SSH key for the git repository.
     """
-    hostname = custom_config_repository.split("@")[1].split("/")[0]
+    splitted_url = custom_config_repository.split("@")
```

Comment:
Maybe using [urlparse](https://docs.python.org/es/3/library/urllib.parse.html) instead? You will have to manually split the "path" part for the branch, but the rest should be simpler.

Reply:
It's not an URL, the branch is prefixed by "@"

Reply:
I don't see a lot of difference, tbh
```
splitted_url = custom_config_repository.split("@")
url = "@".join(splitted_url[0:2])
hostname = parse(url).hostname
branch = splitted_url[2] if len(splitted_url) > 2 else None
```
instead of
```
splitted_url = custom_config_repository.split("@")
url = "@".join(splitted_url[:-1])
hostname = splitted_url[1].split("/")[0]
branch = splitted_url[2] if len(splitted_url) > 2 else None
```

Reply:
Would you be able to add a small comment on the format of `custom_config_repository` please?
I think it would help for next time if something changes :D
```suggestion
# the format of the repository is in git+ssh@<url>:<blahblah>, i.e. <example format>
```

Reply:
I was thinking in something like this
```
url = urlparse("git+ssh://local.juju_model_name@git.launchpad.net/wazuh-server-configuration@main")
hotname = url.hostname
branch = parts[1] if len(parts := url.path.split("@")) > 1 else None
```

or 
```
url = urlparse("git+ssh://local.juju_model_name@git.launchpad.net/wazuh-server-configuration@main")
hotname = url.hostname
try:
   branch =  url.path.split("@")[1]
except  IndexError:
  branch = None
```

Reply:
@javierdelapuente and the URL without the branch I need for later? I don't see that any more simple than it already is
@yanksyoon , I've added it to the docstring

Reply:
The url can also be reconstructed with something like (my goal is to depend on split the minimum):
```
from urllib.parse import urlsplit, urlunsplit
url = urlsplit("git+ssh://local.juju_model_name@git.launchpad.net/wazuh-server-configuration@main")
hostname = url.hostname
branch = parts[1] if len(parts := url.path.split("@")) > 1 else None
newurl = urlunsplit(url._replace(path=url.path.split("@")[0]))
```
or
```
url = urlsplit("git+ssh://local.juju_model_name@git.launchpad.net/wazuh-server-configuration@main")
hostname = url.hostname
path_parts = url.path.split("@"))
branch = path_parts[1] if len(path_parts) > 1 else None
newurl = urlunsplit(url._replace(path=path_parts[0]))
```